rootfs/scheduler/resources/pod.py

```diff
@@ -255,6 +255,14 @@ def _set_resources(self, container, kwargs):
             if resources:
                 container["resources"] = dict(resources)
 
+    def _get_termination_grace_period(self, kwargs):
+        """ return termination grace period """
+        app_type = kwargs.get("app_type")
+        timeout_global = kwargs.get('pod_termination_grace_period_seconds', 30)
+        timeout_local = kwargs.get("pod_termination_grace_period_each", {}).get(app_type)
```

Comment:
it should be, at least it shouldn't behave worse than before because here https://github.com/teamhephy/controller/blob/master/rootfs/scheduler/resources/pod.py#L100 `app_type` is also extracted and `_get_termination_grace_period` is called several lines below from the same method (it is only the place where is it called from). Also we do not modify `kwargs` between call of the my method and place I mentioned above.

Reply:
You are right, as long as `kwargs` are not modified, this should be behave correctly.